### ISSUE TITLE
Cop idea: replace consecutive variable reassignment with chained method calls

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#14490]
This Style cop would detect simple cases where consecutive variable reassignments can be replaced with chained method calls:

```ruby
# bad
name = 'foo'
name = name.upcase
name = name.reverse
name = name.concat('bar')

# good
name = 'foo'.upcase.reverse.concat('bar')
```

Having it all in a single method chain is more readable and allows other cops to further lint/optimize the expression.

**Comment by koic** [comment]
This is a case where the merits should be discussed in the Ruby Style guide before implementing a cop. Please open a proposal in the Ruby Style Guide first:
https://github.com/rubocop/ruby-style-guide
